ntbs-service/Jobs/UnmatchedLabResultAlertsJob.cs

```diff
@@ -31,24 +31,24 @@ public async Task Run(IJobCancellationToken token)
             var dbPotentialMatches =
                 await _specimenService.GetAllSpecimenPotentialMatchesAsync();
 
-            var currentUnmatchedLabResultAlerts =
+            var currentOpenUnmatchedLabResultAlerts =
                 await _alertRepository.GetAllOpenUnmatchedLabResultAlertsAsync();
 
-            var unneededAlerts = currentUnmatchedLabResultAlerts.Where(
+            var unneededAlerts = currentOpenUnmatchedLabResultAlerts.Where(
                 alert => !dbPotentialMatches.Any(dbMatch =>
                     dbMatch.NotificationId == alert.NotificationId &&
                     dbMatch.ReferenceLaboratoryNumber == alert.SpecimenId)).ToList();
 
             Log.Debug($"Number of redundant unmatched lab result alerts to be closed: {unneededAlerts.Count}");
             await _alertRepository.CloseAlertRangeAsync(unneededAlerts);
 
-            var dbMatchesRequiringAlerts = dbPotentialMatches.Where(
-                dbMatch => !currentUnmatchedLabResultAlerts.Any(alert =>
+            var dbMatchesWithoutOpenAlert = dbPotentialMatches.Where(
+                dbMatch => !currentOpenUnmatchedLabResultAlerts.Any(alert =>
                     alert.NotificationId == dbMatch.NotificationId &&
                     alert.SpecimenId == dbMatch.ReferenceLaboratoryNumber)).ToList();
```

Comment:
There were two ways that I could have fixed this bug, and both came with potential performance issues - the first would have been to change this method so that we fetch all of the unmatched alerts. However, the problem with this is that the performance would gradually erode over time as more and more historic closed alerts are returned.

The second approach (that I went with) checks for each potential match whether an alert already exists or not. This also comes with an overhead, but it's one that shouldn't get worse over time.